src/base_client.py

```
@@ -0,0 +1,102 @@
+        Raises:
+            ValueError: For invalid method
+            RuntimeError: For network or API errors
+        """
+        url = f"{self.base_url}/{endpoint}"
+        
+        # Default headers
+        request_headers = headers or {}
+        request_headers.setdefault('Accept', 'application/json')
+        
+        # Log request details
+        self.logger.info(f"Sending {method} request to {url}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)